lib/spack/spack/test/util/git.py

```diff
@@ -1,15 +1,82 @@
 # Copyright Spack Project Developers. See COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from typing import Optional
+
+import pytest
+
 from llnl.util.filesystem import working_dir
 
-from spack.util.git import get_modified_files
+import spack.util.executable as exe
+import spack.util.git
+
+
+def test_git_not_found(monkeypatch):
+    def _mock_find_git() -> Optional[str]:
+        return None
+
+    monkeypatch.setattr(spack.util.git, "_find_git", _mock_find_git)
+
+    git = spack.util.git.git(required=False)
+    assert git is None
+
+    with pytest.raises(exe.CommandNotFoundError):
+        spack.util.git.git(required=True)
 
 
 def test_modified_files(mock_git_package_changes):
     repo, filename, commits = mock_git_package_changes
 
     with working_dir(repo.packages_path):
-        files = get_modified_files(from_ref="HEAD~1", to_ref="HEAD")
+        files = spack.util.git.get_modified_files(from_ref="HEAD~1", to_ref="HEAD")
         assert len(files) == 1
         assert files[0] == filename
+
+
+def test_init_git_repo(git, tmp_path):
+    repo_url = "https://github.com/spack/spack.git"
+    destination = tmp_path / "test_git_init"
+
+    with working_dir(destination, create=True):
+        spack.util.git.init_git_repo(repo_url)
+
+        "No commits yet" in git("status", output=str)
```

Comment:
these and other tests are missing `assert` :)

Reply:
https://github.com/spack/spack/pull/50910